packages/custom-elements/src/CustomElementRegistry.js

```diff
@@ -84,11 +90,64 @@ export default class CustomElementRegistry {
       new DocumentConstructionObserver(internals, document) : undefined;
   }
 
+  /**
+   * @param {string} localName
+   * @param {function(): function(new: HTMLElement)} constructorGetter
+   */
+  polyfillDefineLazy(localName, constructorGetter) {
+    if (!(constructorGetter instanceof Function)) {
+      throw new TypeError('Custom element constructor getters must be functions.');
+    }
+
+    if (!Utilities.isValidCustomElementName(localName)) {
+      throw new SyntaxError(`The element name '${localName}' is not valid.`);
+    }
+
+    if (this.internal_localNameToDefinition(localName)) {
+      throw new Error(`A custom element with name '${localName}' has already been defined.`);
+    }
+
+    if (this._elementDefinitionIsRunning) {
+      throw new Error('A custom element is already being defined.');
+    }
```

Comment:
Seems like these three precondition checks can be made into a utility function called by both `define` and `polyfillDefineLazy`?
* `The element name '${localName}' is not valid.`
* `A custom element with name '${localName}' has already been defined.`
* `A custom element is already being defined.`

Reply:
SGTM, these checks are now their own function.